core/src/test/java/com/radixdlt/api/system/HealthHandlerTest.java

```diff
@@ -64,17 +64,97 @@
 
 package com.radixdlt.api.system;
 
+import static com.radixdlt.lang.Tuple.tuple;
+import static java.util.Objects.requireNonNull;
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.AbstractModule;
 import com.google.inject.Inject;
 import com.radixdlt.api.SystemApiTestBase;
 import com.radixdlt.api.system.generated.models.HealthResponse;
+import com.radixdlt.api.system.generated.models.SignalledReadinessPendingProtocolUpdateState;
 import com.radixdlt.api.system.routes.HealthHandler;
+import com.radixdlt.consensus.bft.Round;
+import com.radixdlt.consensus.epoch.EpochRound;
+import com.radixdlt.monitoring.InMemorySystemInfo;
+import com.radixdlt.monitoring.LedgerSummary;
+import com.radixdlt.protocol.ProtocolUpdateEnactmentCondition;
+import com.radixdlt.protocol.ProtocolUpdateTrigger;
+import com.radixdlt.rev2.Decimal;
+import com.radixdlt.statecomputer.ProtocolState;
+import com.radixdlt.utils.UInt64;
+import java.util.Set;
 import org.junit.Test;
 
 public class HealthHandlerTest extends SystemApiTestBase {
+
+  // Test setup parameters
+  // Threshold: 70% for 20 epochs.
+  // At epoch 100 we already have 5 started epochs of support.
+  // Expected enactment at start of epoch 116.
+  private static final long REQUIRED_EPOCHS = 20L;
+  private static final long STARTED_EPOCHS_OF_SUPPORT = 5L;
+  private static final long CURRENT_EPOCH = 100L;
+  private static final long EPOCH_TARGET_DURATION_MS = 5 * 60 * 1000; /* 5 minutes */
+  private static final long EPOCH_EFFECTIVE_START_MS = 1709835872431L;
+
+  // Test assertion parameters
+  private static final long EXPECTED_PROJECTED_ENACTMENT_EPOCH = 116;
+  private static final long EXPECTED_PROJECTED_ENACTMENT_TIMESTAMP =
+      EPOCH_EFFECTIVE_START_MS + (EPOCH_TARGET_DURATION_MS * 16);
+
   @Inject private HealthHandler sut;
 
+  public HealthHandlerTest() {
+    super(
+        new AbstractModule() {
```

Comment:
The `Modules.override()` and the mocks here look rather cumbersome for testing :(
At the same time, I agree that the "forecast" calculations are worth testing.

How about you extract the calculations out from `*Handler`, into a specialized `*Forecaster` class, and cover it with a unit test instead?